prometheus_metrics.go

```diff
@@ -81,7 +81,27 @@ func (client *MetricsClient) BucketMetrics(ctx context.Context) ([]*prom2json.Fa
 	}
 
 	// Execute GET on /minio/v2/metrics/bucket
-	resp, err := client.executeRequest(ctx, reqData)
+	resp, err := client.executeGetRequest(ctx, reqData)
+	if err != nil {
+		return nil, err
+	}
+	defer closeResponse(resp)
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	return parsePrometheusResults(io.LimitReader(resp.Body, metricsRespBodyLimit))
+}
+
+// ResourceMetrics - returns Resource Metrics in Prometheus format
+func (client *MetricsClient) ResourceMetrics(ctx context.Context) ([]*prom2json.Family, error) {
```

Comment:
The code for each of these metrics functions seems almost same, except for the `relativePath`. It can be put into a common function, something like:

```go
// FetchMetrics - returns Metrics of given subsystem in Prometheus format
func (client *MetricsClient) FetchMetrics(ctx context.Context, subSys string) ([]*prom2json.Family, error) {
	reqData := metricsRequestData{
		relativePath: "/v2/metrics/" + subSys,
	}

	// Execute GET on /minio/v2/metrics/<subSys>
	resp, err := client.executeGetRequest(ctx, reqData)
	if err != nil {
		return nil, err
	}
	defer closeResponse(resp)

	if resp.StatusCode != http.StatusOK {
		return nil, httpRespToErrorResponse(resp)
	}

	return parsePrometheusResults(io.LimitReader(resp.Body, metricsRespBodyLimit))
}
```

Reply:
@anjalshireesh  I like it, I've done it, thanks.